cinnamon-dynamic-wallpaper@TobiZog/files/cinnamon-dynamic-wallpaper@TobiZog/5.4/extension.js

```diff
@@ -249,74 +311,27 @@ CinnamonDynamicWallpaperExtension.prototype = {
 	 * Callback for changes in preferences
 	 */
 	updateLocation: function () {
+		// Update the update information
+		lastLocationUpdate = new Date()
+
 		if (this.autolocation) {
 			let loc = location.estimateLocation()
 			this.latitude = loc["latitude"]
 			this.longitude = loc["longitude"]
-		} else {
-			this.latitude = this.latitude
-			this.longitude = this.longitude
-		}
-
-		// Update the update information
-		lastLocationUpdate = new Date()
-	},
-
-
-	/**
-	 * Main loop
-	 */
-	_loop: function () {
-		if (looping) {
-			this.setImageToTime()
-
-			if (lastLocationUpdate < new Date().getTime() - this.locationRefreshTime * 1000) {
-				this.updateLocation()
-				lastLocationUpdate = new Date()
-			}
 
-			// Refresh every 60 seconds
-			Mainloop.timeout_add_seconds(60, Lang.bind(this, this._loop));
+			this.etrLastUpdate = lastLocationUpdate.getHours() + ":" + lastLocationUpdate.getMinutes()
 		}
-	},
-
-
-	/******************** UI Callbacks ********************/
 
-	/**
-	 * Callback for settings-schema
-	 * Opens the external image configurator window
-	 */
-	openImageConfigurator: function () {
-		Util.spawnCommandLine("/usr/bin/env python3 " +
-			DIRECTORY.path + "/image-configurator/image-configurator.py");
+		this.writeToLogs("Location updated")
 	},
 
-
 	/**
-	 * Callback for settings-schema
-	 * Opens the browser and navigates to the URL of the respository
-	 */
-	openRepoWebsite: function () {
-		Util.spawnCommandLine("xdg-open https://github.com/TobiZog/cinnamon-dynamic-wallpaper");
-	},
-
-
-	/**
-	 * Callback for settings-schema
-	 * Opens the browser and navigates to the URL of the Cinnamon Spices extension
-	 */
-	openSpicesWebsite: function () {
-		Util.spawnCommandLine("xdg-open https://cinnamon-spices.linuxmint.com/extensions/view/97")
-	},
-
-
-	/**
-	 * Callback for settings-schema
-	 * Opens the browser and navigates to the GitHub issue page
+	 * Adding text to the logs
+	 * 
+	 * @param {string} msg New message string
 	 */
-	openIssueWebsite: function () {
-		Util.spawnCommandLine("xdg-open https://github.com/TobiZog/cinnamon-dynamic-wallpaper/issues/new")
+	writeToLogs: function(msg) {
+		this.tvLogs = communication.createLogs(this.tvLogs, msg)
 	}
```

Comment:
It might make sense to add a debug flag here and not default to having it on to prevent spurious logging for users.